libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1749,3 +1745,172 @@ Type const& IRGeneratorForStatements::type(Expression const& _expression)
 	solAssert(_expression.annotation().type, "Type of expression not set.");
 	return *_expression.annotation().type;
 }
+
+bool IRGeneratorForStatements::visit(TryStatement const& _tryStatement)
+{
+	solAssert(!m_returnInfo.has_value(), "");
+
+	_tryStatement.externalCall().accept(*this);
+	solAssert(m_returnInfo.has_value(), "");
+	solAssert(m_returnInfo.value().functionCall == &_tryStatement.externalCall(), "");
+
+	ReturnInfo const returnInfo = move(*m_returnInfo);
+	m_returnInfo.reset();
+
+	string const trySuccessCondition = m_context.trySuccessConditionVariable(_tryStatement.externalCall());
+
+	m_code << "switch iszero(" << trySuccessCondition << ")\n";
+
+	m_code << "case 0 { // success case\n";
+	TryCatchClause const& successClause = *_tryStatement.clauses().front();
+	decodeTryCallReturnParameters(successClause, returnInfo);
+	successClause.block().accept(*this);
+	m_code << "}\n";
+
+	m_code << "default { // failure case\n";
+	handleCatch(_tryStatement);
+	m_code << "}\n";
+
+	return false;
+}
+
+string IRGeneratorForStatements::decodeReturnParameters(ReturnInfo const& _returnInfo)
+{
+	auto const retVars = IRVariable(*_returnInfo.functionCall).commaSeparatedList();
+	if (retVars.empty())
+		return "";
+
+	return Whiskers(R"(
+		let <retVars> := <decodeReturnParameters>(<pos>, <returnSize>)
```

Comment:
Can this use `define()`?

Reply:
I actually think you can use `define()` here and get rid of the function `IRGeneratorForStatements::decodeReturnParameters` altogether.